ls1intum/Artemis
# `Exam mode`: Show exercise group name as exercise name

### Checklist
#### General


#### Server
#### Client


### Motivation and Context


@krusche requested that Artemis shows the exercise group name to all students instead of the exercise name as this faciliates the exam creation for instructors because they do not need to make sure that all exercises have the same title, so all students see the same exercise name. Being able to use different names for the exercises also helps teaching assistants to distinguish the exercises when navigating in Artemis for the exam correction.

### Description
Changed all occurences of `exercise.title` to `exercise.exercisegroup.title` in the exam-submission components and on the exam-overview page.
Do no longer filter out the exercise group if the user is not an instructor on the server side.

Show a message to the instructor that the exercise group name is now shown to all students.
Added a tooltip in the exercise groups overview next to the name to explain the behavior again.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Student
1. Log in to Artemis
2. Navigate to Course Administration
3. Create an exam with all exercise types, when creating the exercise groups make sure that the explanation is understandable.
4. Participate in the exam as a student and make sure that for every exercise the exercise group name is shown.

#### Exam Mode Testing
Covered with the steps above.
### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage







### Screenshots



https://github.com/ls1intum/Artemis/assets/84102468/15bd2aad-5b7d-429e-8203-df2647f6bfb3
